packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -0,0 +1,233 @@
+import * as React from 'react';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Collapse from '@mui/material/Collapse';
+import Container from '@mui/material/Container';
+import Divider from '@mui/material/Divider';
+import Drawer from '@mui/material/Drawer';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import ListSubheader from '@mui/material/ListSubheader';
+import Stack from '@mui/material/Stack';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { styled } from '@mui/material';
+import ToolpadLogo from './ToolpadLogo';
+import { BrandingContext, Navigation, NavigationContext, NavigationPageItem } from '../AppProvider';
+
+const IS_CLIENT = typeof window !== 'undefined';
+
+const DEFAULT_DRAWER_WIDTH = 320;
+
+const LogoContainer = styled('div')({
+  position: 'relative',
+  height: 40,
+  img: {
+    maxHeight: 40,
+  },
+});
+
+interface DashboardSidebarSubNavigationProps {
+  subNavigation: Navigation;
+  depth?: number;
+}
+
+function DashboardSidebarSubNavigation({
+  subNavigation,
+  depth = 0,
+}: DashboardSidebarSubNavigationProps) {
+  // Rerender once to update with client-side navigation
+  const [hasMounted, setHasMounted] = React.useState(false);
```

Comment:
I'm not a big fan of this method. This will prevent rendering certain elements on the first render, even if it's never being hydrated. Have you tried the `useIsSsr` hook that's available in `@mui/utils`?